proto/storage/process_listening_on_port.proto

```diff
@@ -13,51 +13,54 @@ package storage;
+    string pod_id                                  = 4;
+    string pod_uid                                 = 5;
+    ProcessSignal signal                           = 6;
+    string cluster_id                              = 7;
+    string namespace                               = 8;
 
     google.protobuf.Timestamp container_start_time = 9;
 
     string image_id                                = 10;
 }
 
 message ProcessListeningOnPortFromSensor {
-    uint32 port = 1;
-    L4Protocol protocol = 2;
-    ProcessIndicatorUniqueKey process = 3;
+    uint32 port                               = 1;
+    L4Protocol protocol                       = 2;
+    ProcessIndicatorUniqueKey process         = 3;
     google.protobuf.Timestamp close_timestamp = 4;
-    string cluster_id = 6;
-    string deployment_id = 7;
+    string cluster_id                         = 6;
+    string deployment_id                      = 7;
+    string pod_uid                            = 8;
 }
 
 message ProcessListeningOnPortStorage {
     // Ideally it has to be GENERATED ALWAYS AS IDENTITY, which will make it a
     // bigint with a sequence. Unfortunately at the moment some bits of store
     // generator assume an id has to be a string.
-    string id = 1 [(gogoproto.moretags) = 'sql:"pk,type(uuid)"'];
+    string id                         = 1 [(gogoproto.moretags) = 'sql:"pk,type(uuid)"'];
 
-    uint32 port = 2 [(gogoproto.moretags) = 'search:"Port,store"'];
-    L4Protocol protocol = 3 [(gogoproto.moretags) = 'search:"Port Protocol,store"'];
+    uint32 port                       = 2 [(gogoproto.moretags) = 'search:"Port,store"'];
+    L4Protocol protocol               = 3 [(gogoproto.moretags) = 'search:"Port Protocol,store"'];
     google.protobuf.Timestamp close_timestamp = 4 [(gogoproto.moretags) = 'search:"Closed Time,hidden"'];
 
-    string process_indicator_id = 5
+    string process_indicator_id       = 5
         [(gogoproto.moretags) = 'search:"Process ID,store" sql:"fk(ProcessIndicator:id),no-fk-constraint,index=btree,type(uuid)"'];
 
     // XXX: Make it a partial index on only active, not closed, PLOP
-    bool closed = 6 [(gogoproto.moretags) = 'search:"Closed,store" sql:"index=btree"'];
+    bool closed                       = 6 [(gogoproto.moretags) = 'search:"Closed,store" sql:"index=btree"'];
 
     // ProcessIndicator will be not empty only for those cases when we were not
     // able to find references process in the database
     ProcessIndicatorUniqueKey process = 7;
 
-    string deployment_id = 8 [(gogoproto.moretags) = 'search:"Deployment ID,store" sql:"fk(Deployment:id),no-fk-constraint,index=btree,type(uuid)"'];
+    string deployment_id              = 8 [(gogoproto.moretags) = 'search:"Deployment ID,store" sql:"fk(Deployment:id),no-fk-constraint,index=btree,type(uuid)"'];
+
+    string pod_uid                    = 9 [(gogoproto.moretags) = 'search:"Pod UID,hidden" sql:"index=hash,type(uuid)"'];
```

Comment:
Should this be an fk to the pod id column?

Reply:
technically, probably not a bad idea.  Just have to make sure to use the `no-fk-constraint` tag

Reply:
I made the following change

```
string pod_uid                    = 9 [(gogoproto.moretags) = 'search:"Pod UID,hidden" sql:"fk(Pod:id),no-fk-constraint,index=hash,type(uuid)"'];
```

When I run `make generated-srcs` I get the following error

```
pg-table-bindings-wrapper --type=storage.ProcessListeningOnPortStorage --references storage.ProcessIndicator,storage.Deployment --table=listening_endpoints --search-category PROCESS_LISTENING_ON_PORT
2023-10-11 17:51:14 Generating for storage.ProcessListeningOnPortStorage
pkg/postgres/walker: 2023/10/11 10:51:14.508336 schema.go:360: Panic: Couldn't resolve reference in field &{Schema:0xc0004fb700 Name:PodUid ProtoBufName:pod_uid ObjectGetter:{variable:false value:GetPodUid()} ColumnName:PodUid Type:string DataType:string SQLType:uuid ModelType:string Options:{ID:false Ignored:false Index:[0xc00097c400] PrimaryKey:false Unique:false IgnorePrimaryKey:false IgnoreUniqueConstraint:false IgnoreSearchLabels:map[] Reference:0xc00097c3c0 ColumnType:uuid IgnoreChildFKs:false IgnoreChildIndexes:false} Search:{FieldName:Pod UID Enabled:true Ignored:false} DerivedSearchFields:[] Derived:false} (ref: {Pod id true false false <nil> }): type not provided
panic: Couldn't resolve reference in field &{Schema:0xc0004fb700 Name:PodUid ProtoBufName:pod_uid ObjectGetter:{variable:false value:GetPodUid()} ColumnName:PodUid Type:string DataType:string SQLType:uuid ModelType:string Options:{ID:false Ignored:false Index:[0xc00097c400] PrimaryKey:false Unique:false IgnorePrimaryKey:false IgnoreUniqueConstraint:false IgnoreSearchLabels:map[] Reference:0xc00097c3c0 ColumnType:uuid IgnoreChildFKs:false IgnoreChildIndexes:false} Search:{FieldName:Pod UID Enabled:true Ignored:false} DerivedSearchFields:[] Derived:false} (ref: {Pod id true false false <nil> }): type not provided

goroutine 1 [running]:
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc000dc9970, {0x0, 0x0, 0x0})
	/home/jvirtane/go/pkg/mod/github.com/stackrox/zap@v1.15.1-0.20230918235618-2bd149903d0e/zapcore/entry.go:230 +0x486
go.uber.org/zap.(*SugaredLogger).log(0xc000919b80, 0x4, {0x283f8b9?, 0xc00032b900?}, {0xc0009a7b60?, 0xc00032b900?, 0xc00097c3c0?}, {0x0, 0x0, 0x0})
	/home/jvirtane/go/pkg/mod/github.com/stackrox/zap@v1.15.1-0.20230918235618-2bd149903d0e/sugar.go:234 +0x13b
go.uber.org/zap.(*SugaredLogger).Panicf(...)
	/home/jvirtane/go/pkg/mod/github.com/stackrox/zap@v1.15.1-0.20230918235618-2bd149903d0e/sugar.go:159
github.com/stackrox/rox/pkg/logging.(*LoggerImpl).Panicf(0x25acdc0?, {0x283f8b9?, 0x2?}, {0xc0009a7b60?, 0x2?, 0x2540cf9?})
	/home/jvirtane/go/src/github.com/stackrox/stackrox/pkg/logging/logger.go:109 +0x48
github.com/stackrox/rox/pkg/postgres/walker.(*Schema).ResolveReferences(0xc0004fb700, 0xc000e27108)
	/home/jvirtane/go/src/github.com/stackrox/stackrox/pkg/postgres/walker/schema.go:360 +0x205
main.parseReferencesAndInjectPeerSchemas(0x0?, {0xc0009a61a0, 0x2, 0x13?})
	/home/jvirtane/go/src/github.com/stackrox/stackrox/tools/generate-helpers/pg-table-bindings/funcs.go:40 +0x3d7
main.main.func1(0xc00055af00, {0x2735f5b, 0x8, 0x8})
	/home/jvirtane/go/src/github.com/stackrox/stackrox/tools/generate-helpers/pg-table-bindings/main.go:247 +0x653
github.com/spf13/cobra.(*Command).execute(0xc00055af00, {0xc00012e010, 0x8, 0x8})
	/home/jvirtane/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc00055af00)
	/home/jvirtane/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/home/jvirtane/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
main.main()
	/home/jvirtane/go/src/github.com/stackrox/stackrox/tools/generate-helpers/pg-table-bindings/main.go:384 +0x550
exit status 2
central/processlisteningonport/store/postgres/gen.go:3: running "pg-table-bindings-wrapper": exit status 1
```

Reply:
In the `gen.go` you have to update the `--references storage.ProcessIndicator,storage.Deployment`.  Should be as simple as adding `storage.Pod` to the end

Reply:
Done. I remember now, we had a similar issue before and had to make a similar modification.

Reply:
Something is still wrong. There is a failing unit test

```
2023-10-11T22:05:04.2143609Z PASS
2023-10-11T22:05:04.2144848Z 	github.com/stackrox/rox/migrator/migrations/m_189_to_m_190_vulnerability_requests_add_name	coverage: 77.8% of statements
2023-10-11T22:05:04.2314977Z ok  	github.com/stackrox/rox/migrator/migrations/m_189_to_m_190_vulnerability_requests_add_name	0.968s	coverage: 77.8% of statements
2023-10-11T22:05:10.6227461Z pkg/postgres/walker: 2023/10/11 22:05:10.621896 schema.go:360: Panic: Couldn't resolve reference in field &{Schema:0xc001474300 Name:PodUid ProtoBufName:pod_uid ObjectGetter:{variable:false value:GetPodUid()} ColumnName:PodUid Type:string DataType:string SQLType:uuid ModelType:string Options:{ID:false Ignored:false Index:[0xc001c0c680] PrimaryKey:false Unique:false IgnorePrimaryKey:false IgnoreUniqueConstraint:false IgnoreSearchLabels:map[] Reference:0xc001c0c640 ColumnType:uuid IgnoreChildFKs:false IgnoreChildIndexes:false} Search:{FieldName:Pod UID Enabled:true Ignored:false} DerivedSearchFields:[] Derived:false} (ref: {Pod id true false false <nil> }): type not provided
2023-10-11T22:05:10.6258937Z panic: Couldn't resolve reference in field &{Schema:0xc001474300 Name:PodUid ProtoBufName:pod_uid ObjectGetter:{variable:false value:GetPodUid()} ColumnName:PodUid Type:string DataType:string SQLType:uuid ModelType:string Options:{ID:false Ignored:false Index:[0xc001c0c680] PrimaryKey:false Unique:false IgnorePrimaryKey:false IgnoreUniqueConstraint:false IgnoreSearchLabels:map[] Reference:0xc001c0c640 ColumnType:uuid IgnoreChildFKs:false IgnoreChildIndexes:false} Search:{FieldName:Pod UID Enabled:true Ignored:false} DerivedSearchFields:[] Derived:false} (ref: {Pod id true false false <nil> }): type not provided
2023-10-11T22:05:10.6260292Z 
2023-10-11T22:05:10.6260629Z goroutine 1 [running]:
2023-10-11T22:05:10.6266669Z go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc001466dc0, {0x0, 0x0, 0x0})
2023-10-11T22:05:10.6267636Z 	/go/pkg/mod/github.com/stackrox/zap@v1.15.1-0.20230918235618-2bd149903d0e/zapcore/entry.go:230 +0x634
2023-10-11T22:05:10.6268242Z go.uber.org/zap.(*SugaredLogger).log(0xc00034d6a0, 0x4, {0x454b184, 0x44}, {0xc0009623a0, 0x2, 0x2}, {0x0, 0x0, 0x0})
2023-10-11T22:05:10.6362669Z 	/go/pkg/mod/github.com/stackrox/zap@v1.15.1-0.20230918235618-2bd149903d0e/sugar.go:234 +0x18f
2023-10-11T22:05:10.6363220Z go.uber.org/zap.(*SugaredLogger).Panicf(...)
2023-10-11T22:05:10.6367770Z 	/go/pkg/mod/github.com/stackrox/zap@v1.15.1-0.20230918235618-2bd149903d0e/sugar.go:159
2023-10-11T22:05:10.6368272Z github.com/stackrox/rox/pkg/logging.(*LoggerImpl).Panicf(0xc000966810, {0x454b184, 0x44}, {0xc0009623a0, 0x2, 0x2})
2023-10-11T22:05:10.6368672Z 	/__w/stackrox/stackrox/pkg/logging/logger.go:109 +0x91
2023-10-11T22:05:10.6369094Z github.com/stackrox/rox/pkg/postgres/walker.(*Schema).ResolveReferences(0xc001474300, 0xc00112fab0)
2023-10-11T22:05:10.6369626Z 	/__w/stackrox/stackrox/pkg/postgres/walker/schema.go:360 +0x395
2023-10-11T22:05:10.6370073Z github.com/stackrox/rox/migrator/migrations/m_190_to_m_191_plop_add_closed_time_column/schema.glob..func1()
2023-10-11T22:05:10.6370600Z 	/__w/stackrox/stackrox/migrator/migrations/m_190_to_m_191_plop_add_closed_time_column/schema/listening_endpoints.go:34 +0x1f1
2023-10-11T22:05:10.6371092Z github.com/stackrox/rox/migrator/migrations/m_190_to_m_191_plop_add_closed_time_column/schema.init()
2023-10-11T22:05:10.6371761Z 	/__w/stackrox/stackrox/migrator/migrations/m_190_to_m_191_plop_add_closed_time_column/schema/listening_endpoints.go:40 +0x25
2023-10-11T22:05:10.6372279Z FAIL	github.com/stackrox/rox/migrator/migrations/m_190_to_m_191_plop_add_closed_time_column	0.331s
```

I see the following 

```
        // ListeningEndpointsSchema is the go schema for table `listening_endpoints`.
        ListeningEndpointsSchema = func() *walker.Schema {
                schema := walker.Walk(reflect.TypeOf((*storage.ProcessListeningOnPortStorage)(nil)), "listening_endpoints")
                referencedSchemas := map[string]*walker.Schema{
                        "storage.ProcessIndicator": referenceschema2.ProcessIndicatorsSchema,
                        "storage.Deployment":       referenceschema2.DeploymentsSchema,
                }

                schema.ResolveReferences(func(messageTypeName string) *walker.Schema {
                        return referencedSchemas[fmt.Sprintf("storage.%s", messageTypeName)]
                })
                schema.SetOptionsMap(search.Walk(v1.SearchCategory_PROCESS_LISTENING_ON_PORT, "processlisteningonportstorage", (*storage.ProcessListeningOnPortStorage)(nil)))
                schema.ScopingResource = resources.DeploymentExtension
                return schema
        }()
```

here https://github.com/stackrox/stackrox/blob/0e9cb1dd2d8a6dafa86031bf2fb3e76591801a62/migrator/migrations/m_190_to_m_191_plop_add_closed_time_column/schema/listening_endpoints.go#L27C1-L40C5

I think I need to add a line like
```
"storage.Pod":              referenceschema2.PodSchema,
```
However the corresponding file does not exist. It should be automatically generated, but it has not been. Which command should generate it?

Reply:
I have run `make proto-generated-srcs` and `make generated-srcs` without error.

Reply:
This is working now.